### DISCUSSION TITLE
Dataview

### DISCUSSION THREAD

**Discussion by dixtel** [Feel-ix-343/markdown-oxide#198] [Q&A] [answered]
Hey Feel-ix-343, its is possible to implement dataview to render the results of the query?. I dont know how it will look likes it the terminal editor. I know that in the web-based obsidian client it is rendered as html and can be accessible by mouse. How it will work in the terminal based text editor environment?

Maybe its is stupid question but I haven't done any projects related to LSP and Text Editors like helix.

**Comment by Feel-ix-343** [comment, marked as answer]
Hey! This has been an idea for a while; the way to do this would be to make a dataview execution engine and then implement the UX through LSP references: calling lsp references on a dataview block would run the query and return the locations. 

I think this would be pretty cool, but there are other priorities for me and I now have limited time. Generally I think that vector search for like 90% of usecases and I generally view these things as not being worth the time. However, deterministic organization definitely has its role (tasks, synthesizing organized knowledge, ...) so I totally see the appeal of this. I'd welcome and review any PRs for this.

**Reply by dixtel** [reply]
Yeah, I like the idea to have PKM in the my favourite text editor. Currently I'm using official obsidian client to organize my work (my tickets, knowledge about domains, useful information related to my work). Its a bit of pain to navigate thorough my notes using mouse and doing modification without helix motions.

I will try to learn the codebase of this project and do some useful work sometimes during my limited free time :crying_cat_face:

**Reply by Feel-ix-343** [reply]
ok cool! yea im sure you will get 90% optimal utility out of dataview just using it in the obsidian UI anyway
